src/Parser/Monk/Brewmaster/Modules/Features/StaggerPoolGraph.js

```diff
@@ -0,0 +1,276 @@
+      this._purifyTimestamps.push(event.timestamp);
+    }
+  }
+
+  on_toPlayer_death(event) {
+    this._deathEvents.push(event);
+  }
+
+  on_toPlayer_damage(event) {
+    this._hpEvents.push(event);
+  }
+
+  on_toPlayer_heal(event) {
+    this._hpEvents.push(event);
+  }
+
+  plot() {
+    // x indices
+    const labels = Array.from({length: Math.ceil(this.owner.fightDuration / 1000)}, (x, i) => i);
+
+    // somethingBySeconds are all objects mapping from seconds ->
+    // something, where if a value is unknown for that timestamp it is
+    // undefined (the key is still present)
+    const poolBySeconds = labels.reduce((obj, sec) => {
+      obj[sec] = undefined;
+      return obj;
+    }, {});
+    const hpBySeconds = Object.assign({}, poolBySeconds);
+
+    const purifies = this._purifyTimestamps.map(timestamp => Math.floor((timestamp - this.owner.fight.start_time) / 1000) - 1);
+    const deaths = this._deathEvents.map(({ timestamp, killingAbility }) => {
+      return { 
+        seconds: Math.floor((timestamp - this.owner.fight.start_time) / 1000),
+        ability: killingAbility,
+      };
+    });
+
+    this._staggerEvents.forEach(({ timestamp, newPooledDamage }) => {
+      const seconds = Math.floor((timestamp - this.owner.fight.start_time) / 1000);
+      // for simplicity, we plot the right-edge of each bin. stagger
+      // actually ticks twice a second
+      poolBySeconds[seconds] = newPooledDamage;
+    });
+
+    this._hpEvents.forEach(({ timestamp, hitPoints, maxHitPoints }) => {
+      const seconds = Math.floor((timestamp - this.owner.fight.start_time) / 1000);
+      // we fill in the blanks later if hitPoints is not defined
+      if(!!hitPoints) {
+        hpBySeconds[seconds] = { hitPoints, maxHitPoints };
+      }
+    });
+
+    // fill in blanks. after deaths hp and stagger should be set to
+    // zero. in periods of no activity, the same hp/stagger should be
+    // preserved.
+    //
+    // you might wonder "but i thought stagger ticked twice a second? so
+    // either you're dead or damage is incoming!" Friendo, let me
+    // introduce you to the joy of stagger pausing: BoC + ISB pauses the
+    // dot for 3 seconds so we may have gaps where we are *not* dead and
+    // also not taking damage
+    let lastPoolContents = 0;
+    let lastHpContents = { hitPoints: 0, maxHitPoints: 0 };
+    for(const label in labels) {
+      if(poolBySeconds[label] === undefined) {
+        poolBySeconds[label] = lastPoolContents;
+      } else {
+        lastPoolContents = poolBySeconds[label];
+      }
+
+      if(hpBySeconds[label] === undefined) {
+        hpBySeconds[label] = lastHpContents;
+      } else {
+        lastHpContents = hpBySeconds[label];
+      }
+
+      if(!!deaths.find(event => event.seconds === Number(label))) {
+        lastPoolContents = 0;
+        lastHpContents = { hitPoints: 0, maxHitPoints: lastHpContents.maxHitPoints };
```

Comment:
When a person dies, doesn't it make sense to set his maximum HP to 0 as well? Considering you're doing this as a default action anyway (when there's no information about HP *and there was none before to copy*).